resources/qml/Settings/SettingView.qml

```diff
@@ -9,229 +9,406 @@ import QtQuick.Layouts 1.1
-            onExpandedChanged: Printer.setExpandedCategories(expanded)
-            onVisibilityChanged: Cura.SettingInheritanceManager.forceUpdate()
+        filterContainer.visible = !filterContainer.visible
+        if(filterContainer.visible)
+        {
+            filter.forceActiveFocus();
+        }
+        else
+        {
+            filter.text = "";
         }
+    }
+
+    Rectangle
+    {
+        id: filterContainer
+        visible: false
 
-        delegate: Loader
+        border.width: UM.Theme.getSize("default_lining").width
+        border.color:
         {
-            id: delegate
-
-            width: UM.Theme.getSize("sidebar").width;
-            height: provider.properties.enabled == "True" ? UM.Theme.getSize("section").height : - contents.spacing
-            Behavior on height { NumberAnimation { duration: 100 } }
-            opacity: provider.properties.enabled == "True" ? 1 : 0
-            Behavior on opacity { NumberAnimation { duration: 100 } }
-            enabled:
+            if(mouseArea.containsMouse || clearFilterButton.containsMouse)
             {
-                if(!ExtruderManager.activeExtruderStackId && ExtruderManager.extruderCount > 0)
-                {
-                    // disable all controls on the global tab, except categories
-                    return model.type == "category"
-                }
-                return provider.properties.enabled == "True"
+                return UM.Theme.getColor("setting_control_border_highlight");
+            }
+            else
+            {
+                return UM.Theme.getColor("setting_control_border");
             }
+        }
 
-            property var definition: model
-            property var settingDefinitionsModel: definitionsModel
-            property var propertyProvider: provider
-            property var globalPropertyProvider: inheritStackProvider
+        color: UM.Theme.getColor("setting_control")
 
-            //Qt5.4.2 and earlier has a bug where this causes a crash: https://bugreports.qt.io/browse/QTBUG-35989
-            //In addition, while it works for 5.5 and higher, the ordering of the actual combo box drop down changes,
-            //causing nasty issues when selecting different options. So disable asynchronous loading of enum type completely.
-            asynchronous: model.type != "enum" && model.type != "extruder"
-            active: model.type != undefined
+        anchors
+        {
+            top: parent.top
+            left: parent.left
+            leftMargin: UM.Theme.getSize("default_margin").width
+            right: parent.right
+            rightMargin: UM.Theme.getSize("default_margin").width
+        }
+        height: visible ? UM.Theme.getSize("setting_control").height : 0
+        Behavior on height { NumberAnimation { duration: 100 } }
 
-            source:
+        TextField
+        {
+            id: filter;
+            visible: !monitoringPrint
```

Comment:
Unnecessary, right? Its parent is `filterContainer`, which should also be invisible then if I'm not mistaken.